app.rb

```diff
@@ -25,37 +26,49 @@ def initialize
     @bucket = @s3.bucket(settings.aws_bucket)
   end
 
-  get '/' do
-    if authed?
-      erb :index, locals: { projects: key_list }
-    else
-      redirect '/auth/myusa'
-    end
-  end
-
   get '/auth/myusa/callback' do
     halt(401, 'Not Authorized') unless env['omniauth.auth']
     session[:user_email] = env['omniauth.auth'].info.email
     redirect '/'
   end
 
+  before %r{^(?!\/auth)} do
+    redirect '/auth/myusa' unless authed?
+  end
```

Comment:
Was going to make a comment in the last pull request that "we should have it redirect by default", but wasn't sure of the best way to do so in Sinatra. This is great!